client/state/data-layer/wpcom/timezones/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import { TIMEZONES_REQUEST } from 'state/action-types';
+
+import {
+	timezonesRequestSuccessAction,
+	timezonesRequestFailureAction,
+	timezonesReceiveAction,
+} from 'state/timezones/actions';
+
+const undocumented = wpcom.undocumented();
+
+/*
+ * Start a request to WordPress.com server to get the timezones data
+ */
+export const requestTimezones = ( { dispatch } ) => (
+	undocumented
+		.timezones()
+		.then( zones => {
+			dispatch( timezonesRequestSuccessAction() );
+			dispatch( timezonesReceiveAction( zones ) );
```

Comment:
yeah I've had this concern for a while but unlike scattered redux-thunk action creators the middleware gives us one centralized and clear place to put them. when I first started doing this with the importer it ended up saving me considerable headaches because as the API changed (it did) and as we better understood the process (we learned) there was a single spot to update the code and nothing broke.

this is the power of abstracting at the data object level which Calypso expects rather than at the API level.